src/components/GoalsList/GoalsList.js

```diff
@@ -0,0 +1,104 @@
+
+    useEffect(() => {
+        getGoals();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    const updateGoal = (goal) => {
+        setLoading(true);
+        axios
+            .put(`${process.env.REACT_APP_SERVER_URL}/api/goal/${goal.id}/update`, {
+                status: goal.status,
+                topicId: goal.topicId,
+                userId: goal.userId
+            })
+            .then(res => {
+                console.log(res.data);
+                getGoals();
+            })
+            .catch(err => {
+                setAlert({ open: true, message: err.response.data.message, severity: 'error' });
+                setLoading(false);
+            });
+    };
+
+    const deleteGoal = (goal) => {
+        setLoading(true);
+        axios
+            .delete(`${process.env.REACT_APP_SERVER_URL}/api/goal/${goal.id}/delete`)
+            .then(res => {
+                console.log(res.data);
+                getGoals();
+            })
+            .catch(err => {
+                setAlert({ open: true, message: err.response.data.message, severity: 'error' });
+                setLoading(false);
+            });
+    };
+
+    const getGoals = () => {
+        setLoading(true);
+        axios
+            .get(`${process.env.REACT_APP_SERVER_URL}/api/goal/${user.id}/list`)
+            .then(res => {
+                setGoals(res.data);
+                console.log(res.data);
+                setLoading(false);
+            })
+            .catch(err => {
+                setAlert({ open: true, message: err.response.data.message, severity: 'error' });
+                setLoading(false);
+            });
+    };
+
+    const addGoal = (goal) => {
+        setLoading(true);
+        axios
+            .post(`${process.env.REACT_APP_SERVER_URL}/api/goal/add`, { topicId: topic, userId: user.id })
+            .then(res => {
+                getGoals();
+                setAlert({ open: true, message: 'Goal added successfully!', severity: 'success' });
+            })
+            .catch(err => {
+                setLoading(false);
+                setAlert({ open: true, message: err.response.data.message, severity: 'error' });
+            });
+    };
+
+    return (
+        <Grid container spacing={3} direction="row">
+            <Grid item xs={6}>
+                <Typography variant="h4">
+                    Goals
+                    </Typography>
+            </Grid>
+            <Grid item xs={6}>
+                <Button variant="contained" color="primary" onClick={(e) => setNewGoalCard(true)}>Add New Goal</Button>
+            </Grid>
+            <Grid item xs={12}>
+                {newGoalCard ? <NewGoalCard topics={topics} setNewGoalCard={setNewGoalCard} addGoal={addGoal} topic={topic} setTopic={setTopic} /> : ''}
+                {
+                    goals.map(goal =>
+                        <GoalCard goal={goal} updateGoal={updateGoal} deleteGoal={deleteGoal} />
```

Comment:
On goal delete I would suggest to add one more dialog and confirm if user is sure that he wants to delete this goal.